### DISCUSSION TITLE
System.ArgumentException : Unknown framework version 7.0

### DISCUSSION THREAD

**Discussion by Sumanaoka** [nunit/nunit3-vs-adapter#1173] [Q&A]
I have this issue when trying to run a pipeline in Azure DevOps (which I never encounter since my last successful Pipeline run Aug 2023) and take note in my local the project is working fine.
I saw this old post (https://github.com/nunit/nunit3-vs-adapter/issues/987) and at first I tried updating the NUnit3TestAdapter to latest version as mentioned but no luck. I already tried different versions even the latest one but still the same issue. I don't know what to do now as I am working on this for the past 2 weeks.

![image](https://github.com/nunit/nunit3-vs-adapter/assets/120444788/f54a2eae-6754-4398-b6c0-c7936f522167)

In my local I am using the following:
Visual Studio 2019
Target Framework 4.6.1 and tried updating to 4.7 and 4.8 still the same
NUnit version I already tried from 3.9 up to latest versions still the same
NUnit3TestAdapter version I already tried from 3.10 up to latest version still the same

**Comment by OsirisTerje** [comment]
Sorry for seeing this late.

With the latest 4.5 version, you should not see this.  

Can you show your csproj file and pipeline yaml file please.

**Comment by OsirisTerje** [comment]
@Sumanaoka I am sorry, but too little information to say anything.  Suggest you create a small repro project and upload that.  You can then use the same way you use in the real project without exposing anything.

In the yaml, the compile and test step seems to use a custom powershell script  `build.ps1`.  The content of that could reveal something.  

In the csproj file it depends how complicated that has been made.  But, just seeing what packagereferences it has would be a help.  

The best however, is a repro project.

**Comment by Sumanaoka** [comment]
Tried running a test on cmd on my local and here's the result.

![image](https://github.com/nunit/nunit3-vs-adapter/assets/120444788/57001292-9086-40a2-9e62-9cedf8f19cbd)

**Comment by OsirisTerje** [comment]
Ok, so it works then ?

(You do have net 7 installed on your build server?)
